### ISSUE TITLE
conan download --recipe does not download export_source

### ISSUE THREAD

**Issue by antze-k** [conan-io/conan#5193]
Conan version 1.15.0
Python 3.7.3
`uname -srmo` => `Linux 4.19.33-un-def-alt0.M80P.1 x86_64 GNU/Linux`

I am attempting to clone a package from a chosen remote, to have it locally in a "buildable" state. This is not achievable (with exception of dirty hacks mentinoned below).

If I attempt to download only the recipe with `conan download --recipe`, conan does not download the export_source directory, therefore, the downloaded package is not buildable without access to the remote.

Example 1:
```
$ conan remote list
conan-center: https://conan.bintray.com [Verify SSL: True]

$ rm -rf ~/.conan/data

$ conan download zlib/1.2.11@conan/stable --recipe --remote conan-center
( ... successful download output ... )

$ conan copy zlib/1.2.11@conan/stable my/stable
Downloading conan_sources.tgz
[==================================================] 7.1KB/7.1KB      
Decompressing conan_sources.tgz: 100%|██████████| 6.91k/6.91k [00:00<00:00, 1.13MB/s]
Copied zlib/1.2.11@conan/stable to zlib/1.2.11@my/stable
Copied sources zlib/1.2.11@conan/stable to zlib/1.2.11@my/stable
```

(`conan copy` still performs some downloading, while the copy operation is probably assumed to be local)

Example 2:
```
$ rm -rf ~/.conan/data
$ conan download zlib/1.2.11@conan/stable --recipe --remote conan-center
( ... successful download output ... )
$ conan remote remove conan-center
$ conan copy zlib/1.2.11@conan/stable my/stable
ERROR: Error while trying to get recipe sources for zlib/1.2.11@conan/stable. No remote defined
```

This behavior is a bit strange in my opinion, especially considering the fact that documentation explicitly states:

> Using exports_sources the package recipe can be self-contained, containing the source code like in a snapshot, and then not requiring downloading or retrieving the source code from other origins (git, download) with the source() method when it is necessary to build from sources.

and:

> The main difference with exports is that exports files are always retrieved (even if pre-compiled packages exist), while exports_sources files are only retrieved when it is necessary to build a package from sources.

https://docs.conan.io/en/latest/reference/conanfile/attributes.html#exports-sources

Actually it works a kind of opposite, because if I don't specify `--recipe` and try to download any binary package (even non-existent one!) conan DOES download export_source directory:

```
$ conan download zlib/1.2.11@conan/stable --package something-nonexistent --remote conan-center
( ... downloading progress ... )
zlib/1.2.11@conan/stable: Downloading zlib/1.2.11@conan/stable:something-nonexistent
zlib/1.2.11@conan/stable: Retrieving package something-nonexistent from remote 'conan-center' 
ERROR: Binary package not found: 'zlib/1.2.11@conan/stable:something-nonexistent'

$ ls -a ~/.conan/data/zlib/1.2.11/conan/stable
.  ..  export  export_source  metadata.json  metadata.json.lock
```

The only reasonable workaround I've found is:
- `conan download $PACKAGE_ID --package something-nonexistent --remote conan-center || true`
- `mkdir -p ~/.conan/data/PACKAGE/PATH/export_source` (because the packages which don't have `exports_sources` and rely on `source()` method instead don't create the directory, and `conan copy` command breaks in `complete_recipe_sources` method (client/source.py:15) - I will provide the reproduction scenario if requested).

---

- [X] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [X] I've specified the Conan version, operating system version and any tool that can be relevant.
- [X] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by memsharded** [comment]
Yes, I think you are right, and this should be considered a bug.
Proposing this fix https://github.com/conan-io/conan/pull/5194 for the next Conan 1.16 release.

Thanks very much for the detailed report.

**Comment by lasote** [comment]
Will be released on 1.16. Thanks for reporting.
